tests/slow/src/test/scala/tests/mill/MillServerCodeLensSuite.scala

```diff
@@ -40,6 +43,8 @@ class MillServerCodeLensSuite
       _ <- server.initialize()
       _ <- server.initialized()
       _ <- server.executeCommand(ServerCommands.GenerateBspConfig)
+      _ <- server.server.buildServerPromise.future
+        .withTimeout(2, TimeUnit.MINUTES)
```

Comment:
Doesn't `didOpen` wait for the `buildServerPromise` anyway? I though it's more of a when the bsp compiles and publishes things not the connection itself.

Reply:
Might be, but I can't figure out where does the test actually hang :/

Reply:
Maybe add printing of the bsp trace on error. That could provide some insight.